Push/PushConfiguration.swift

```diff
@@ -3,80 +3,63 @@
 // file, You can obtain one at http://mozilla.org/MPL/2.0/
 
 import Foundation
+import MozillaAppServices
 
-public enum PushConfigurationLabel: String {
-    case fennec = "Fennec"
-    case fennecEnterprise = "FennecEnterprise"
-    case firefoxBeta = "FirefoxBeta"
-    case firefoxNightlyEnterprise = "FirefoxNightly"
-    case firefox = "Firefox"
-
-    public func toConfiguration() -> PushConfiguration {
-        switch self {
-        case .fennec: return FennecPushConfiguration()
-        case .fennecEnterprise: return FennecEnterprisePushConfiguration()
-        case .firefoxBeta: return FirefoxBetaPushConfiguration()
-        case .firefoxNightlyEnterprise: return FirefoxNightlyEnterprisePushConfiguration()
-        case .firefox: return FirefoxPushConfiguration()
-        }
-    }
-}
-
-public protocol PushConfiguration {
-    var label: PushConfigurationLabel { get }
-
-    /// The associated autopush server should speak the protocol documented at
-    /// http://autopush.readthedocs.io/en/latest/http.html#push-service-http-api
-    /// /v1/{type}/{app_id}
-    /// type == apns
-    /// app_id == the “platform” or “channel” of development (e.g. “firefox”, “beta”, “gecko”, etc.)
-    var endpointURL: NSURL { get }
-}
-
-public struct FennecPushConfiguration: PushConfiguration {
-    public init() {}
-    public let label = PushConfigurationLabel.fennec
-    public let endpointURL = NSURL(string: "https://updates.push.services.mozilla.com/v1/apns/fennec")!
-}
-
-public struct FennecStagingPushConfiguration: PushConfiguration {
-    public init() {}
-    public let label = PushConfigurationLabel.fennec
-    public let endpointURL = NSURL(string: "https://updates-autopush.stage.mozaws.net/v1/apns/fennec")!
-}
+let PUSH_PROD_HOST = "updates.push.services.mozilla.com"
+let PUSH_STAGE_HOST = "updates-autopush.stage.mozaws.net"
```

Comment:
We try to avoid global constant. We can create a struct to hold it, or put it elsewhere where it makes sense. Also for example `PUSH_PROD_HOST` feels more like an ObjC name. Preferrably would be of format `let pushProdHost = "updates.push.services.mozilla.com"`

Reply:
Aah! good catch, I'm now thinking maybe an enum could satisfy the known push hosts!, i.e:

```swift
enum KnownPushHost: String {
    case prod = "updates.push.services.mozilla.com"
    case stage = "updates-autopush.stage.mozaws.net"
}
```

and maybe put that in `AppConstants.swift`, that seems like a place where the constants go 😬?

Reply:
I'm not a fan of `AppConstants.swift` since it has constants for the whole app - was done before our time. Most of the time constants relates to one area of the codebase only, so unless exception that constant should live in that specific area IMO. Your `enum KnownPushHost:` makes sense to me 👍

Reply:
Alrighty thought about it for a bit and a sturct made a little bit more sense in how I use it 😬 pushed a commit to change it https://github.com/mozilla-mobile/firefox-ios/pull/14649/commits/4c17d3dab10f39e06df9864a931d493e98237259!